bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/handler/ElroConnectsBridgeHandler.java

```diff
@@ -0,0 +1,880 @@
+            case POWERSOCKET:
+                device = new ElroConnectsDevicePowerSocket(deviceId, this);
+                break;
+            case ENTRY_SENSOR:
+                device = new ElroConnectsDeviceEntrySensor(deviceId, this);
+                break;
+            case MOTION_SENSOR:
+                device = new ElroConnectsDeviceMotionSensor(deviceId, this);
+                break;
+            case TH_SENSOR:
+                device = new ElroConnectsDeviceTemperatureSensor(deviceId, this);
+                break;
+            default:
+                logger.debug("Device type {} not supported", deviceType);
+                return null;
+        }
+        device.setDeviceType(deviceType);
+        devices.put(deviceId, device);
+        return device;
+    }
+
+    /**
+     * Just before sending message, this method should be called to make sure we wait for all responses that are still
+     * expected to be received. The last response will indicated by a token in the last response message.
+     *
+     * @param waitResponse true if we want to wait for response for next message to be sent before allowing subsequent
+     *            message
+     */
+    private void awaitResponse(boolean waitResponse) {
+        CompletableFuture<Boolean> waiting = awaitResponse;
+        if (waiting != null) {
+            try {
+                logger.trace("Waiting for previous response before sending");
+                waiting.get(RESPONSE_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+            } catch (InterruptedException | ExecutionException | TimeoutException ignore) {
+                logger.trace("Wait for previous response timed out");
+            }
+        }
+        awaitResponse = waitResponse ? new CompletableFuture<>() : null;
+    }
+
+    /**
+     * This method is called when all responses on a request have been received.
+     */
+    private void stopAwaitResponse() {
+        CompletableFuture<Boolean> future = awaitResponse;
+        if (future != null) {
+            future.complete(true);
+        }
+        awaitResponse = null;
+    }
+
+    private void sendAck(DatagramSocket socket) throws IOException {
+        logger.debug("Send Ack: {}", ACK_STRING);
+        socket.send(ackPacket);
+    }
+
+    private String sendAndReceive(DatagramSocket socket, String query, boolean broadcast)
+            throws UnknownHostException, IOException {
+        send(socket, query, broadcast);
+        byte[] buffer = new byte[4096];
+        DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
+        return receive(socket, buffer, packet);
+    }
+
+    private void send(DatagramSocket socket, String query, boolean broadcast) throws IOException {
+        final InetAddress address = broadcast
+                ? InetAddress.getByName(networkAddressService.getConfiguredBroadcastAddress())
+                : addr;
+        if (address == null) {
+            logger.debug("No address specified, cannot send");
```

Comment:
Should this be a warn with a slightly longer message to guide users to select a network in network settings?